plugins/aws/fix_plugin_aws/resource/inspector.py

```diff
@@ -0,0 +1,446 @@
+    status: Optional[str] = field(default=None, metadata={"description": "The status of the finding."})  # fmt: skip
+    title: Optional[str] = field(default=None, metadata={"description": "The title of the finding."})  # fmt: skip
+    type: Optional[str] = field(default=None, metadata={"description": "The type of the finding. The type value determines the valid values for resource in your request. For more information, see Finding types in the Amazon Inspector user guide."})  # fmt: skip
+    updated_at: Optional[datetime] = field(default=None, metadata={"description": "The date and time the finding was last updated at."})  # fmt: skip
+
+    @classmethod
+    def collect_resources(cls: Type[AwsResource], builder: GraphBuilder) -> None:
+        # Default behavior: in case the class has an ApiSpec, call the api and call collect.
+        log.debug(f"Collecting {cls.__name__} in region {builder.region.name}")
+        if spec := cls.api_spec:
+            try:
+                items = builder.client.list(
+                    aws_service=spec.service,
+                    action=spec.api_action,
+                    result_name=spec.result_property,
+                    expected_errors=spec.expected_errors,
+                    filterCriteria={"awsAccountId": [{"comparison": "EQUALS", "value": f"{builder.account.id}"}]},
+                )
+                cls.collect(items, builder)
+            except Boto3Error as e:
+                msg = f"Error while collecting {cls.__name__} in region {builder.region.name}: {e}"
+                builder.core_feedback.error(msg, log)
+                raise
+            except Exception as e:
+                msg = f"Error while collecting {cls.__name__} in region {builder.region.name}: {e}"
+                builder.core_feedback.info(msg, log)
+                raise
+
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        if finding_resources := self.finding_resources:
+            for finding_resource in finding_resources:
+                if rid := finding_resource.id:
+                    if "arn:aws:lambda" in rid:
+                        # remove lambda's version from arn to connect by arn
+                        lambda_arn = rid.rsplit(":", 1)[0]
+                        builder.add_edge(
+                            self,
+                            clazz=AwsLambdaFunction,
+                            arn=lambda_arn,
+                        )
+                    else:
+                        builder.add_edge(
+                            self,
+                            clazz=(AwsEc2Instance, AwsEcrRepository),
+                            id=rid,
+                        )
```

Comment:
based on the arn, you also know if it is ec2 instance or ecr repo, or?

Reply:
Same logic was in 
```python
class AwsInspectorV2Coverage(AwsResource):
    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
        if self.id:
            builder.add_edge(
                self,
                clazz=AwsEc2Instance,
                id=self.id,
            )
        if resource_metadata := self.resource_metadata:
            if lambda_metadata := resource_metadata.lambda_function:
                builder.add_edge(
                    self,
                    clazz=AwsLambdaFunction,
                    id=lambda_metadata.function_name,
                )
            if ecr_metadata := resource_metadata.ecr_repository:
                builder.add_edge(
                    self,
                    clazz=AwsEcrRepository,
                    id=ecr_metadata.name,
                )
 ```
 For `AwsInspectorFinding` we have this logic:
 If `finding_resource.type` is Ec2 instance, then finding_resource.id is an ID of `Ec2Instance`. If `finding_resource.type` is EcrRepository, then `finding_resource.id` is the name of `EcrRepository`. For lambda `finding_resource.id` will be `arn + lambda_version` of `AwsLambdaFunction`

Reply:
So then: let's define the class to filter more explicitly.